File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nums_dict = {value: index for index, value in enumerate(nums)}

        for i, n in enumerate(nums):
            diff = target - n
            if diff in nums_dict and nums_dict[diff] != i:
                return [i, nums_dict[diff]]
        
        return []

File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::unordered_map<int, int> noDuplicates;

        for (const int& key : nums){
            noDuplicates[key] = 1;
        }
        if (nums.size()!=noDuplicates.size()){
            return true;
        }
        else{
            return false;
        }
    }
};